### PR TITLE
Server side2

### PR DESCRIPTION
**Pull request by seunomonije**
Added support for db storage

### CODE REVIEW

**Comment by seunomonije** [comment]
Forgot to request a review yesterday smh

File: portfolio/pom.xml [hunk 1]
```diff
@@ -30,8 +30,17 @@
...
+    <!-- App Engine dependency for Datastore-->
+    <dependency>
+        <groupId>com.google.appengine</groupId>
+        <artifactId>appengine-api-1.0-sdk</artifactId>
+        <version>1.9.59</version>
+    </dependency>
   </dependencies>
 
+
+
```
Comment (jessicaslaughter) [inline portfolio/pom.xml]: nit: remove extra lines

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -35,38 +46,60 @@ public void init(){
...
-        String json = gson.toJson(arr);
-        return json;
-    }
-
     @Override
     public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException { 
         String input = getParameter(request, "text-input", "");
+        long timestamp = System.currentTimeMillis();
+
+        messages.add(input); //adds to array
```
Comment (jessicaslaughter) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: the comment here is probably not necessary since the code is self-explanatory

**Review by jessicaslaughter — CHANGES_REQUESTED** [review]

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -35,38 +46,60 @@ public void init(){
 
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {  
+        // Send a query for the comments, latest first
+        Query query = new Query("comments").addSort("timestamp", SortDirection.DESCENDING);
```
Comment (jpnickolas) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Line is longer than 80 characters. Please format to be under 80 characters.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 4]
```diff
@@ -35,38 +46,60 @@ public void init(){
...
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        PreparedQuery results = datastore.prepare(query);
+
+        for (Entity entity: results.asIterable()) {
+            long id = entity.getKey().getId();
+            String message = (String)entity.getProperty("text");
+            long timestamp = (long)entity.getProperty("timestamp");
```
Comment (jpnickolas) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: `id` and `timestamp` are unused here.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 5]
```diff
@@ -35,38 +46,60 @@ public void init(){
...
+     * was not specified by the client
+     */
     private String getParameter(HttpServletRequest request, String name, String defaultValue) {
         String value = request.getParameter(name);
         if (value == null) {
-        return defaultValue;
+            return defaultValue;
         }
-        return value;
+            return value;
```
Comment (jpnickolas) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Unindent line

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 6]
```diff
@@ -35,38 +46,60 @@ public void init(){
...
+    }
+
+    private String convertToJsonUsingGson(ArrayList arr) {
+        Gson gson = new Gson();
+        String json = gson.toJson(arr);
+        return json;
     }
+
```
Comment (jpnickolas) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:89]: nit: remove newline

**Review by jpnickolas — CHANGES_REQUESTED** [review]

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 3 above)
Comment (jessicaslaughter) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: (pulling from convo on Ibiyemi's PR) it looks like Java should be under 100 characters (https://google.github.io/styleguide/javaguide.html#s4.4-column-limit)

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 7]
```diff
@@ -12,16 +12,27 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+
```
Comment (jessicaslaughter) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: nit: remove new line

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 8]
```diff
@@ -12,16 +12,27 @@
...
 
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
+import java.util.ArrayList;
+
+
```
Comment (jessicaslaughter) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: nit: remove new line

**Review by jessicaslaughter — CHANGES_REQUESTED** [review]

File: portfolio/pom.xml (hunk 1 above)
Comment (jpnickolas) [reply portfolio/pom.xml]: This was never fixed. Please fix the change before resolving it.

**Review by jpnickolas — CHANGES_REQUESTED** [review]

**Review by jpnickolas — APPROVED** [review]

**Review by jessicaslaughter — APPROVED** [review]

**Pull request merged**